src/RulesDeploy.ts

```diff
@@ -66,7 +77,39 @@ export class RulesDeploy {
         })
       );
     });
-    return Promise.all(promises);
+    return Promise.all(promises).catch(async (err) => {
+      if (err.status === QUOTA_EXCEEDED_STATUS_CODE) {
+        utils.logBullet(
+          clc.bold.yellow(this.type + ":") + " quota exceeded error while uploading rules"
+        );
+
+        const history: ListRulesetsEntry[] = await gcp.rules.listAllRulesets(this.options.project);
+
+        if (history.length > RULESET_COUNT_LIMIT) {
+          const shouldContinue = await prompt.once({
```

Comment:
This should be short-circuited if `--force` flag has been passed.

Reply:
done, not sure exactly how `prompt` is _supposed_ to be used, but this works